### DISCUSSION TITLE
自动获取国内和海外域名通道

### DISCUSSION THREAD

**Discussion by creeksidenetworks** [pymumu/smartdns#1791] [Ideas]
墙内DNS中毒，一般是返回一个海外的IP地址。因此，建议你增加一个功能，这样就不用使用GFW List这种非常笨且不可靠的机制：
1）收到DNS请求，向海外和国内多个DNS服务器发送请求；
2）如果国内DNS返回的是海外IP，弃用结果，等待海外的DNS解析结果；
3）路由器上配置策略路由，只要是国内的IP段，就走本地，否则走隧道。

**Comment by PikuZheng** [comment]
1 smartdns机制本就是这样的
2 针对国内上游配置并启用黑名单即可。按照你的说法，黑名单应该是所有国外ip

**Comment by ixmu** [comment]
你获取到了想要的配置文件了吗？可否分享一下
